### PR TITLE
patch: add patch for Node.js v12.18.4

### PR DESCRIPTION
**Pull request by imballinst**
This PR adds patch for Node.js v12.18.4. The only different part is:

```js
// v12.18.1.
const {
  internalModuleReadJSON,
  internalModuleStat
} = internalBinding('fs');

// v12.18.4.
const { internalModuleStat } = internalBinding('fs');
```

Hence the patch is 1 line fewer than the `node.v12.18.1.cpp.patch` one.

Signed-off-by: Try Ajitiono <ballinst@gmail.com>

### CODE REVIEW

File: patches/patches.json [hunk 1]
```diff
@@ -11,6 +11,9 @@
   "v12.18.1": [
     "node.v12.18.1.cpp.patch"
   ],
+  "v12.18.14": [
+    "node.v12.18.14.cpp.patch"
+  ],
```
Comment (dio) [inline patches/patches.json]:
```suggestion
  "v12.18.4": [
    "node.v12.18.4.cpp.patch"
  ],
```

**Review by dio — COMMENTED** [review]
Have you tried running:

```
$ yarn prepare
$ GITHUB_USERNAME=Imballinst yarn start
```

?

**Comment by imballinst** [comment]
...whoops, you're right, I haven't. I also made an error with the version 😅. Will fix it. Thanks!

**Comment by imballinst** [comment]
Now I can run it, but from what I have read from people's experience, the process could take ages...

```
$ node lib-es5/upload.js
> Building built-v12.18.4-macos-x64...
> Cloning Node.js repository from GitHub...
  git                          [====================] 100%
> Checking out v12.18.4
> Applying patches
> Compiling Node.js from sources...
  make                         [                    ] 0%
```

**Comment by imballinst** [comment]
...and it failed when building the older version.

```
> Cloning Node.js repository from GitHub...
  git                          [====================] 100%
> Checking out v6.17.1
> Applying patches
> Compiling Node.js from sources...
    rc = gyp.main(args)
  File "tools/gyp/pylib/gyp/__init__.py", line 538, in main
    return gyp_main(args)
  File "tools/gyp/pylib/gyp/__init__.py", line 523, in gyp_main
    generator.GenerateOutput(flat_list, targets, data, params)
  File "tools/gyp/pylib/gyp/generator/make.py", line 2174, in GenerateOutput
    part_of_all=qualified_target in needed_targets)
  File "tools/gyp/pylib/gyp/generator/make.py", line 800, in Write
    self.Pchify))
  File "tools/gyp/pylib/gyp/generator/make.py", line 1195, in WriteSources
    cflags = self.xcode_settings.GetCflags(configname)
  File "tools/gyp/pylib/gyp/xcode_emulation.py", line 551, in GetCflags
    archs = self.GetActiveArchs(self.configname)
  File "tools/gyp/pylib/gyp/xcode_emulation.py", line 420, in GetActiveArchs
    xcode_archs_default = GetXcodeArchsDefault()
  File "tools/gyp/pylib/gyp/xcode_emulation.py", line 118, in GetXcodeArchsDefault
    xcode_version, _ = XcodeVersion()
  File "tools/gyp/pylib/gyp/xcode_emulation.py", line 1265, in XcodeVersion
    version = re.match(r'(\d\.\d\.?\d*)', version).groups()[0]
AttributeError: 'NoneType' object has no attribute 'groups'
> Error! Error: ./configure failed with code 1
    at ChildProcess.<anonymous> (/Users/tryajitiono/code/node-pkg-fetch/lib-es5/spawn.js:68:23)
    at ChildProcess.emit (events.js:310:20)
    at maybeClose (internal/child_process.js:1021:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)
error Command failed with exit code 2.
```

**Comment by dio** [comment]
Yeah, I think we can simulate this with a couple of commands, and we need it to work on alpine:

1. Clone nodejs/node
2. Apply the patch
3. Run in an alpine docker image

```
$ ./configure --dest-cpu x64 --without-inspector --without-snapshot --fully-static --enable-static
$ make -j 8 # in a powerful machine
```

**Review by dio — APPROVED** [review]

**Comment by dio** [comment]
Deps to be installed in `alpine:3.10`:

```
apk add --update --no-cache curl \
    make \
    python \
    g++ \
    gcc \
    linux-headers \
    paxctl
```

**Pull request merged**